docs/pages/api/access-plugin.mdx

```diff
@@ -0,0 +1,738 @@
+---
```

Comment:
This guide provides a great minimal example of how to consume incoming events :). It relies on the fact you can walk through all rows in the spreadsheet to find the one to update, making the process stateless.

Real-world plugins almost always have to introduce a stateful component (the message ID the channel was sent to, whether the event was already processed, ...). This allows to increase reliability and run replicated plugins. A step by step how-to about those topics would not be practical nor useful, but it might be useful to mention the teleport pluginData API in the `NextSteps` section. Essentially telling that it exists, uses a compare-and-swap approach to deal with concurrency, and that it allows to process events in a stateful way.

We could also mention that we strongly discourage handling the approval at the plugin level (e.g. with a message answer) because it could lead to severe security issues where an unauthorized person can respond to the plugin and escalate privileges.

Reply:
Thanks! I've added this information